iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/ProjectContext.java

```diff
@@ -19,24 +19,72 @@
  */
 package org.sonar.iac.kubernetes.visitors;
 
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.iac.kubernetes.model.ProjectResource;
+
+import static org.sonar.iac.helm.HelmFileSystem.retrieveHelmProjectFolder;
+
 /**
  * Data class to provide information about the project. This allows to share cross-file knowledge to the individual checks.
  */
 public final class ProjectContext {
+  private final Map<String, Map<String, Set<ProjectResource>>> projectResourcePerNamespacePerPath = new HashMap<>();
+  private final FileSystem fileSystem;
 
-  private ProjectContext() {
+  private ProjectContext(FileSystem fileSystem) {
+    this.fileSystem = fileSystem;
   }
 
-  public static Builder builder() {
-    return new Builder();
+  /**
+   * Get all resources of a given {@code clazz} in a given {@code namespace} and that are accessible to a file with the given {@code path}.
+   * This means that the resources can be in the same file, or in the same directory, or in the descendant directories, but not in the ancestor directories.
+   */
+  public Set<ProjectResource> getProjectResources(String namespace, String path, Class<? extends ProjectResource> clazz) {
+    if (projectResourcePerNamespacePerPath.containsKey(namespace)) {
+      var resourcesPerPath = projectResourcePerNamespacePerPath.get(namespace);
+
+      var filePath = Path.of(path);
+      var basePath = Optional.ofNullable(retrieveHelmProjectFolder(filePath, fileSystem))
```

Comment:
I'm not a fan of calling `retrieveHelmProjectFolder` for every file.
In my opinion this information should be present in the Context.
If it's a `HelmInputFileContext` we should store the HelmProjectFolder once during the construction of the Context and retrieve it from there (there are multiple calls to retrieve the Folder during the whole analysis).
And if it's not a `HelmInputFileContext` we should directly go to `filePath.getParent()`

Reply:
Yes, this was another idea I had. Then I'll go this way and change the HelmInputFileContext